pydantic/utils.py

```diff
@@ -125,7 +127,12 @@ def display_as_type(v):
 
 
 def to_snake_case(v: str) -> str:
-    v = re.sub(r'([A-Z]+)([A-Z][a-z])', r'\1_\2', v)
-    v = re.sub(r'([a-z\d])([A-Z])', r'\1_\2', v)
+    return SNAKE_CASE_REGEX.sub(r'\1\3_\2\4', v).lower()
 
-    return v.replace('-', '_').lower()
+
+@contextmanager
+def change_exception(raise_exc, *except_types):
```

Comment:
Great improvement! But I think better to use another signature for func  `change_exception(raise_exc, class_or_tuple)` like in `isinstance` or `issubclass`, with this more easily to understand what you what to catch and what to raise when use this func.

Reply:
done.

Reply:
You changed only name, not signature

Reply:
Oh, I see what you mean, I've always really disliked the signature of `isinstance` adn `issubclass`, I'd rather keep it like this.

Reply:
Ok :) but better to rename back arg name then

Reply:
yes, I was being dumb.